pages/chaoss/visualizations/contrib_importance_over_time.py

```diff
@@ -0,0 +1,423 @@
+    (df_final['Commit'],
+    df_final['Issue Opened'], 
+    df_final['PR Opened'],
+    df_final['PR Comment'], 
+    df_final['PR Review']) = zip(*df_final.apply(lambda row: cntrb_importance_over_time(df, row.period_from, row.period_to, window_width, threshold), axis=1))
+    
+    return df_final
+
+def create_figure(df_final, step_size):
+    # create plotly express line graph
+    fig = go.Figure(
+        [
+            go.Scatter(name = 'Commit',
+                       x = df_final['period_from'],
+                       y = df_final['Commit'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[0])
+                       ), 
+            
+            go.Scatter(name = 'Issue Opened',
+                       x = df_final['period_from'],
+                       y = df_final['Issue Opened'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[1])
+                       ), 
+
+            go.Scatter(name = 'PR Opened',
+                       x = df_final['period_from'],
+                       y = df_final['PR Opened'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[2])
+                       ),
+            
+            go.Scatter(name = 'PR Request Comment',
+                       x = df_final['period_from'],
+                       y = df_final['PR Comment'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[3])
+                       ),
+
+            go.Scatter(name = 'PR Request Review',
+                       x = df_final['period_from'],
+                       y = df_final['PR Review'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[4])
+                       )
+                        
+        ], 
+    )
+    
+    # define x-axis and y-axis titles and intialize first x-axis tick to start at the user-inputted start_date
+    start_date = min(df_final['period_from'])
+
+    # update xaxes to display ticks
+    fig.update_xaxes(showgrid=True,
+                     ticklabelmode="period",
+                     tickangle=45,  
+                     dtick=f"M{step_size * 2}", 
+                     tickformat='%b %Y',)
+
+    
+    # hover template styling 
+    fig.update_traces(textposition="top right", 
+                      hovertemplate= "Date: %{x}" + "<br>Contributor Importance: %{y}<br><extra></extra>",
+                      )
+    
+    # layout syling
+    fig.update_layout(
+        xaxis_title=f'Timeline (stepsize = {step_size})',
+        xaxis=dict(tick0=start_date),
+        yaxis_title='Contributor Importance', 
```

Comment:
Repeating as I wasnt able to comment on the last one and including message sent via chat: "turn x axis to be parallel to the graph, allows more space for the graph" 

So since the x axis values are at an angle currently they take up more space. Since this graph has so many inputs its size is already a lot smaller than most graphs so anything space saving is good. Honestly since it is already just ending up being "Oct 20xx" ticks with just the year makes sense. Personally I dont care too much as long as they can all fit without being rotated.